dbt/task/generate.py

```diff
@@ -127,6 +128,15 @@ def _get_manifest(self):
         return manifest
 
     def run(self):
+        compile_results = None
+        if self.args.compile:
+            compile_results = super(GenerateTask, self).run()
+            if any(r.errored for r in compile_results):
```

Comment:
can this code path actually be reached? I think dbt fails hard on the first compilation error and (I think?) raises an exception

Reply:
Well, I assumed that it can be reached, because this exists: https://github.com/fishtown-analytics/dbt/blob/9a395facfdc785430b18eec95a507728d64aa42c/dbt/task/base_task.py#L16-L22

And it seems to me that https://github.com/fishtown-analytics/dbt/blob/9a395facfdc785430b18eec95a507728d64aa42c/dbt/node_runners.py#L88-L106 means we can set `error` without raising, if it's an `InternalException` or in `catchable_errors` (so `CompilationException` or `RuntimeException`).